marshal/testSizeCheckUnmarshal/testStructs_test.go

```diff
@@ -0,0 +1,64 @@
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/gogo/protobuf/protobuf  --gogoslick_out=. testStruct.proto
+
+package testSizeCheckUnmarshal
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshlizerProtoObj(t *testing.T) {
+	t.Parallel()
+
+	gm := &marshal.GogoProtoMarshalizer{}
+
+	ts2 := &TestStruct2{
+		Field1: 1,
+		Field2: []byte("field2"),
+		Field3: []byte("field3"),
+	}
+	ts1 := &TestStruct1{}
+
+	marshalizedTs2, _ := gm.Marshal(ts2)
+	err := gm.Unmarshal(ts1, marshalizedTs2)
+	assert.Nil(t, err)
+}
+
+func TestSizeUnmarshalizerTestStruct2ToTestStruct1Err(t *testing.T) {
+	t.Parallel()
+
+	gm := &marshal.GogoProtoMarshalizer{}
+	sizeCheckMarslalizer := marshal.NewSizeCheckUnmarshalizer(gm, 20)
+
+	ts2 := &TestStruct2{
+		Field1: 1,
+		Field2: []byte("field2"),
+		Field3: []byte("field3"),
+	}
+	ts1 := &TestStruct1{}
+
+	marshalizedTs2, _ := sizeCheckMarslalizer.Marshal(ts2)
+	err := sizeCheckMarslalizer.Unmarshal(ts1, marshalizedTs2)
+	assert.NotNil(t, err)
```

Comment:
can you verify the error is ErrUnmarshallingBadSize instead?

Reply:
done.